src/Tizen.NUI/src/internal/FrameBroker/FrameData.cs

```diff
@@ -0,0 +1,382 @@
+        /// </summary>
+        internal ImageView Image
+        {
+            get
+            {
+                if (_image == null)
+                {
+                    _image = new ImageView();
+                    renderer = new Renderer(CreateQuadGeometry(), CreateShader());
+                    textureSet = new TextureSet();
+                }
+                switch (Type)
+                {
+                    case FrameType.RemoteSurfaceTbmSurface:
+                        if (TbmSurface == null)
+                        {
+                            return null;
+                        }
+                        textureSet.SetTexture(0, new Texture(TbmSurface));
+                        renderer.SetTextures(textureSet);
+                        _image.AddRenderer(renderer);
+                        break;
+                    default:
+                        break;
+                }
+
+                return _image;
+            }
+        }
+
+        /// <summary>
+        /// Checks whether the direction of the frame is forward or not.
+        /// </summary>
+        internal bool DirectionForward
+        {
+            get
+            {
+                Interop.FrameBroker.FrameDirection direction = Interop.FrameBroker.FrameDirection.Backward + 1;
+                Interop.FrameBroker.ErrorCode err = Interop.FrameBroker.GetDirection(_frame, out direction);
+                if (err != Interop.FrameBroker.ErrorCode.None)
+                {
+                    Log.Error(LogTag, "Failed to get direction");
+                }
+                return (direction == Interop.FrameBroker.FrameDirection.Forward);
+            }
+        }
+
+        /// <summary>
+        /// Gets the extra data.
+        /// </summary>
+        internal Bundle Extra
+        {
+            get
+            {
+                SafeBundleHandle safeBundle;
+                Interop.FrameBroker.ErrorCode err = Interop.FrameBroker.GetExtraData(_frame, out safeBundle);
+                if (err != Interop.FrameBroker.ErrorCode.None)
+                {
+                    Log.Error(LogTag, "Failed to get extra data");
+                    return null;
+                }
+                return new Bundle(safeBundle);
+            }
+        }
+
+        /// <summary>
+        /// Enumeration for the frame type.
+        /// </summary>
+        internal enum FrameType
+        {
+            /// <summary>
+            /// The tbm surface of the remote surface.
+            /// </summary>
+            RemoteSurfaceTbmSurface,
+
+            /// <summary>
+            /// The image file of the remote surface.
+            /// </summary>
+            RemoteSurfaceImageFile,
+
+            /// <summary>
+            /// The image of the splash screen.
+            /// </summary>
+            SplashScreenImage,
+
+            /// <summary>
+            /// The edje of the splash screen.
+            /// </summary>
+            SPlashScreenEdje,
```

Comment:
does this mean Edge?

Reply:
Edje guide : https://docs.tizen.org/iot/api/5.0/tizen-iot-headed/edje_main.html
FrameBroker support you draw the screen in four types.
This type is received from native code. We can also receive edje type in NUI.
FRAME_TYPE_REMOTE_SURFACE_NATIVE_BUFFER: The remote surface - The native buffer
FRAME_TYPE_REMOTE_SURFACE_IMAGE_FILE: The remote surface - The image file
FRAME_TYPE_SPLASH_SCREEN_IMAGE: The splash screen - The image
FRAME_TYPE_SPLASH_SCREEN_EDJE: The splash screen - The EDJE